llmfoundry/utils/config_utils.py

```diff
@@ -51,6 +53,39 @@ def update_batch_size_info(cfg: DictConfig):
     return cfg
 
 
+def process_init_device(model_cfg: DictConfig, fsdp_config: dict):
+    # Restrict model init_device to 'meta' and 'cpu',
+    # using 'cuda' vs. 'cuda:id' is tricky and can lead to common user errors
+    # when multiple GPUs are available.
+    # Also 'meta' is only valid when using FSDP
+    init_context = contextlib.nullcontext()
+    if 'init_device' in model_cfg:
+        assert model_cfg.init_device in ['meta', 'cpu', 'mixed']
+        if fsdp_config is None and model_cfg.init_device == 'meta':
+            warnings.warn(
+                "Using `cfg.model.init_device='meta'` is only valid when using FSDP! " +\
+                "Reverting to `cfg.model.init_device='cpu'`.")
+            model_cfg.init_device = 'cpu'
+        if model_cfg.init_device == 'meta':
+            init_context = init_empty_weights()
```

Comment:
Hey there! I noticed that when running the quick start example here: https://docs.mosaicml.com/projects/mcli/en/latest/guides/first_llm.html the run fails because `init_empty_weights` is not defined 
```
----------Begin global rank 7 STDERR----------
╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /llm-foundry/scripts/train/train.py:328 in <module>                          │
│                                                                              │
│   325 │   │   yaml_cfg = om.load(f)                                          │
│   326 │   cli_cfg = om.from_cli(args_list)                                   │
│   327 │   cfg = om.merge(yaml_cfg, cli_cfg)                                  │
│ ❱ 328 │   main(cfg)                                                          │
│   329                                                                        │
│                                                                              │
│ /llm-foundry/scripts/train/train.py:203 in main                              │
│                                                                              │
│   200 │   │   cfg.pop('fsdp_config')                                         │
│   201 │   │   fsdp_config = None                                             │
│   202 │                                                                      │
│ ❱ 203 │   init_context = process_init_device(cfg.model, fsdp_config)         │
│   204 │                                                                      │
│   205 │   # build tokenizer                                                  │
│   206 │   tokenizer = build_tokenizer(cfg.tokenizer)                         │
│                                                                              │
│ /llm-foundry/llmfoundry/utils/config_utils.py:70 in process_init_device      │
│                                                                              │
│   67 │   │   │   │   "Reverting to `cfg.model.init_device='cpu'`.")          │
│   68 │   │   │   model_cfg.init_device = 'cpu'                               │
│   69 │   │   if model_cfg.init_device == 'meta':                             │
│ ❱ 70 │   │   │   init_context = init_empty_weights()                         │
│   71 │   │   if model_cfg.init_device == 'mixed':                            │
│   72 │   │   │   if fsdp_config is None:                                     │
│   73 │   │   │   │   raise NotImplementedError(                              │
╰──────────────────────────────────────────────────────────────────────────────╯
NameError: name 'init_empty_weights' is not defined
```
Just wanted to flag it :)

Reply:
Thank you!